dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -153,30 +155,34 @@ public class PagedDoraWorker extends AbstractWorker implements DoraWorker {
    * Constructor.
    *
    * @param workerId
+   * @param identity
    * @param conf
    * @param cacheManager
    * @param membershipManager
    */
   @Inject
   public PagedDoraWorker(
       @Named("workerId") AtomicReference<Long> workerId,
+      WorkerIdentity identity,
       AlluxioConfiguration conf,
       CacheManager cacheManager,
       MembershipManager membershipManager
   ) {
-    this(workerId, conf, cacheManager, membershipManager, new BlockMasterClientPool(),
+    this(workerId, identity, conf, cacheManager, membershipManager, new BlockMasterClientPool(),
         FileSystemContext.create(conf));
   }
 
   protected PagedDoraWorker(
       AtomicReference<Long> workerId,
+      WorkerIdentity identity,
       AlluxioConfiguration conf,
       CacheManager cacheManager,
       MembershipManager membershipManager,
       BlockMasterClientPool blockMasterClientPool,
       FileSystemContext fileSystemContext) {
     super(ExecutorServiceFactories.fixedThreadPool("dora-worker-executor", 5));
     mWorkerId = workerId;
+    mWorkerIdentity = identity;
```

Comment:
so for the transition period, we would have both AtomicReference<Long> workerId and WorkerIdentity identity? if not for non-disruptive upgrade consideration, wouldn't be enough just need WorkerIdentity alone which already has a long-representing WorkerIdentity who can recognize master passed long-type id is it?

Reply:
Yes, this should replace the atomic long.

Reply:
I spent a few moments looking into replacing the long-based ID with the new worker identity, but it turned out non-trivial. Let's focus on the implementation of the new worker identity first in this PR, and do the replacement in follow-up PRs.